### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by cramertj**

### CODE REVIEW

File: futures-channel/benches/sync_mpsc.rs [hunk 1]
```diff
@@ -108,71 +107,67 @@ struct TestSender {
...
-        try_ready!(self.tx.poll_ready(cx).map_err(|_| ()));
-        self.tx.start_send(self.last + 1).unwrap();
-        self.last += 1;
-        assert!(self.tx.poll_flush(cx).unwrap().is_ready());
-        Ok(Async::Ready(Some(self.last)))
+
+    fn poll_next(mut self: PinMut<Self>, cx: &mut task::Context)
+        -> Poll<Option<Self::Item>>
+    {
+        let this = &mut *self;
```
Comment (MajorBreakfast) [inline futures-channel/benches/sync_mpsc.rs]: Why?

File: futures-channel/benches/sync_mpsc.rs [hunk 2]
```diff
@@ -108,71 +107,67 @@ struct TestSender {
...
-            assert_eq!(Ok(Async::Ready(Some(i + 1))), tx.poll_next(&mut cx));
-            assert_eq!(Ok(Async::Pending), tx.poll_next(&mut cx));
-            assert_eq!(Ok(Async::Ready(Some(i + 1))), rx.poll_next(&mut cx));
-        }
+    noop_cx(|cx| {
+        b.iter(|| {
+            let (tx, mut rx) = channel(0);
+
+            let mut tx = TestSender {
+                tx: tx,
```
Comment (MajorBreakfast) [inline futures-channel/benches/sync_mpsc.rs]: clippy would complain

File: futures-channel/benches/sync_mpsc.rs (hunk 1 above)
Comment (cramertj) [reply futures-channel/benches/sync_mpsc.rs]: I need to access `self.tx` and `self.last` in overlapping regions.

File: futures-executor/benches/poll.rs [hunk 3]
```diff
@@ -28,27 +30,27 @@ fn task_init(b: &mut Bencher) {
...
     };
+    impl Unpin for MyFuture {}
 
     impl Future for MyFuture {
-        type Item = ();
-        type Error = ();
+        type Output = ();
 
-        fn poll(&mut self, cx: &mut task::Context) -> Poll<(), ()> {
+        fn poll(mut self: PinMut<Self>, cx: &mut task::Context) -> Poll<Self::Output> {
```
Comment (MajorBreakfast) [inline futures-executor/benches/poll.rs]: I see that before it was also `task::Context`, but the others are imported directly

File: futures-executor/benches/thread_notify.rs [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
 extern crate futures;
 extern crate futures_executor;
 extern crate test;
 
+use std::marker::Unpin;
+use std::mem::PinMut;
+
 use futures::prelude::*;
 use futures::task::{self, Waker};
 use futures_executor::block_on;
```
Comment (MajorBreakfast) [inline futures-executor/benches/thread_notify.rs]: `futures::executor`

File: futures-executor/benches/thread_notify.rs [hunk 5]
```diff
@@ -19,23 +22,22 @@ fn thread_yield_single_thread_one_wait(b: &mut Bencher) {
...
-                Ok(Async::Pending)
+                Poll::Pending
             }
         }
     }
 
     b.iter(|| {
         let y = Yield { rem: NUM };
-        block_on(y).unwrap();
+        block_on(y);
```
Comment (MajorBreakfast) [inline futures-executor/benches/thread_notify.rs]: Why did you remove that? Is there no warning?

File: futures-channel/benches/sync_mpsc.rs (hunk 1 above)
Comment (MajorBreakfast) [reply futures-channel/benches/sync_mpsc.rs]: ah interesting. The `unsafe_unpinned` macros also have a similar problem I think

File: futures-channel/benches/sync_mpsc.rs (hunk 2 above)
Comment (cramertj) [reply futures-channel/benches/sync_mpsc.rs]: Done.
